Sources/PackageLoading/PackageDescription4Loader.swift

```diff
@@ -342,6 +343,16 @@ extension PackageDependencyDescription {
         let requirement = try Requirement(v4: json.get("requirement"))
         let location = try fixLocation(json.get("url"), requirement: requirement)
         let name: String? = json.get("name")
+        
+        if let validPath = try? AbsolutePath(validating: location), fileSystem.exists(validPath) {
```

Comment:
@neonichu had some good points about making this even more generic, we should adopt that approach

Reply:
https://github.com/apple/swift-package-manager/pull/3236#discussion_r572457452

Reply:
Yeah, we'll work to make it generic to support other forms of source control

Reply:
I think @neonichu idea was to define the method in the abstract protocol `RepositoryProvider` say `RepositoryProvider::isValidDirectory` or `RepositoryProvider::validateDirectory(path:) throws` (or some other better name) so that this would work for other types of repository providers - we have more than just Git!

Reply:
this would mean we would need to inject the RepositoryProvider into this method, so that would require some research on how to best do that plumbing cc @neonichu